### PR TITLE
update TFOptimizer doc

### PR DESCRIPTION
**Pull request by yangw1234**

### CODE REVIEW

File: docs/docs/APIGuide/TFPark/tf-optimizer.md [hunk 1]
```diff
@@ -0,0 +1,126 @@
+TFOptimizer is used for optimizing a TensorFlow model with respect to its training variables
+on Spark/BigDL.
+
+**Create a TFOptimizer**:
+```python
+import tensorflow as tf
+from zoo.tfpark import TFOptimizer
+optimizer = TFOptimizer.from_loss(loss, Adam(1e-3))
```
Comment (hkvision) [inline docs/docs/APIGuide/TFPark/tf-optimizer.md]: missing import Adam?

File: docs/docs/APIGuide/TFPark/tf-optimizer.md [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+#### Arguments
+
+
+* **loss**: The loss tensor of the TensorFlow model, should be a scalar
+        
+* **optim_method**: the optimization method to be used, such as bigdl.optim.optimizer.Adam
+* **session**: the current TensorFlow Session, if you want to used a pre-trained model,
+             you should use the Session to load the pre-trained variables and pass it to TFOptimizer.
+* **val_outputs**: the validation output TensorFlow tensor to be used be val_methods
+* **val_labels**: the validation label TensorFlow tensor to be used be val_methods
```
Comment (hkvision) [inline docs/docs/APIGuide/TFPark/tf-optimizer.md]: be used be? or be used for?

File: docs/docs/APIGuide/TFPark/tf-optimizer.md [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+               this value.
+* **clip_value**: float >= 0. Gradients will be clipped when their absolute value
+                exceeds this value.
+* **metrics**: a dictionary. The key should be a string representing the metric's name
+             and the value should be the corresponding TensorFlow tensor, which should be a scalar.
+* **tensor_with_value**: a dictionary. The key is TensorFlow tensor, usually a
+                      placeholder, the value of the dictionary is a tuple of two elements. The first one of
+                      the tuple is the value to feed to the tensor in training phase and the second one
+                      is the value to feed to the tensor in validation phase.
+* **kwargs**: used for backward compatibility
```
Comment (hkvision) [inline docs/docs/APIGuide/TFPark/tf-optimizer.md]: what does this mean?

File: docs/docs/APIGuide/TFPark/tf-optimizer.md [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+### from_keras (factory method)
+
+```python
+from_keras(keras_model, dataset, optim_method=None, val_spilt=0.0, **kwargs)
+```
+
+#### Arguments
+
+* **keras_model**: the tensorflow.keras model, which must be compiled.
+* **dataset**: a TFDataset
```
Comment (hkvision) [inline docs/docs/APIGuide/TFPark/tf-optimizer.md]: Add link to TFDataSet?

File: docs/docs/APIGuide/TFPark/tf-optimizer.md [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+
+## Methods
+
+### from_loss (factory method)
+
+```python
+from_loss(loss, optim_method, session=None, val_outputs=None,
+                  val_labels=None, val_method=None, val_split=0.0,
+                  clip_norm=None, clip_value=None, metrics=None,
+                  tensor_with_value=None, **kwargs)
```
Comment (hkvision) [inline docs/docs/APIGuide/TFPark/tf-optimizer.md]: For `from_loss`, users may be confusing where to use TFDataSet (since from_keras has argument dataset), maybe we need some more documentations or link to some existing example?

File: docs/docs/APIGuide/TFPark/tf-optimizer.md [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+```python
+set_train_summary(summary)
+```
+
+#### Arguments
+
+* **summary**: The train summary to be set. A TrainSummary object contains information
+               necessary for the optimizer to know how often the logs are recorded,
+               where to store the logs and how to retrieve them, etc. For details,
+               refer to the docs of bigdl.optim.optimizer.TrainSummary.
```
Comment (hkvision) [inline docs/docs/APIGuide/TFPark/tf-optimizer.md]: Directly link to BigDL document page?

File: docs/docs/APIGuide/TFPark/tf-optimizer.md [hunk 7]
```diff
@@ -0,0 +1,126 @@
...
+
+### optimize
+
+```python
+optimize(self, end_trigger=None)
+```
+
+#### Arguments
+
+* **end_trigger**: BigDL's Trigger to indicate when to stop the training. If none, defaults to
```
Comment (hkvision) [inline docs/docs/APIGuide/TFPark/tf-optimizer.md]: Link to BigDL's document page?

File: docs/mkdocs.yml [hunk 8]
```diff
@@ -63,6 +63,7 @@ pages:
   - TFPark API:
     - KerasModel: APIGuide/TFPark/model.md
     - TFEstimator: APIGuide/TFPark/estimator.md
+    - TFOptimizer: APIGuide/TFPark/tf-optimizer.md
```
Comment (hkvision) [inline docs/mkdocs.yml]: What about TFPredictor?

File: docs/docs/APIGuide/TFPark/tf-optimizer.md (hunk 1 above)
Comment (yangw1234) [reply docs/docs/APIGuide/TFPark/tf-optimizer.md]: added

File: docs/docs/APIGuide/TFPark/tf-optimizer.md (hunk 2 above)
Comment (yangw1234) [reply docs/docs/APIGuide/TFPark/tf-optimizer.md]: used by, fixed

File: docs/docs/APIGuide/TFPark/tf-optimizer.md (hunk 3 above)
Comment (yangw1234) [reply docs/docs/APIGuide/TFPark/tf-optimizer.md]: removed this

File: docs/docs/APIGuide/TFPark/tf-optimizer.md (hunk 4 above)
Comment (yangw1234) [reply docs/docs/APIGuide/TFPark/tf-optimizer.md]: added

File: docs/docs/APIGuide/TFPark/tf-optimizer.md (hunk 5 above)
Comment (yangw1234) [reply docs/docs/APIGuide/TFPark/tf-optimizer.md]: add more description

File: docs/docs/APIGuide/TFPark/tf-optimizer.md (hunk 6 above)
Comment (yangw1234) [reply docs/docs/APIGuide/TFPark/tf-optimizer.md]: added

File: docs/docs/APIGuide/TFPark/tf-optimizer.md (hunk 7 above)
Comment (yangw1234) [reply docs/docs/APIGuide/TFPark/tf-optimizer.md]: added
